bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieRequestProcessor.java

```diff
@@ -564,10 +564,19 @@ public void operationComplete(Future<Channel> future) throws Exception {
                     AuthHandler.ServerSideHandler authHandler = c.pipeline()
                             .get(AuthHandler.ServerSideHandler.class);
                     authHandler.authProvider.onProtocolUpgrade();
-                    if (future.isSuccess()) {
+
+                    /*
+                     * Success of the future doesn't guarantee success in authentication
+                     * future.isSuccess() only checks if the result field is not null
+                     */
+                    if (future.isSuccess() && authHandler.authenticated) {
```

Comment:
can we add a "isAuthenticated" method to authHandler ?
this way we want access directly the field

Reply:
Fixed in [7276947](https://github.com/apache/bookkeeper/commit/72769477f52e4982081ae87a8e43da0a0e34e2d1)